pandas/tests/indexes/test_common.py

```diff
@@ -391,3 +398,40 @@ def test_astype_preserves_name(self, index, dtype):
             assert result.names == index.names
         else:
             assert result.name == index.name
+
+
+@pytest.mark.parametrize("na_position", [None, "middle"])
+def test_sort_values_invalid_na_position(index_with_missing, na_position):
+
+    if type(index_with_missing) in [DatetimeIndex, PeriodIndex, TimedeltaIndex]:
+        pytest.xfail("stable descending order sort not implemented")
```

Comment:
e.g. maybe just enumerate the PeriodIndex case here (do the others have an issue?) pls open if they don't (you can make a single issue with checkboxes)